pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1499,6 +1535,31 @@ func (n *NetworkPolicyController) syncAppliedToGroup(key string) error {
 	return nil
 }
 
+// getAppliedToWorkloads returns a list of workloads like Pods and ExternalEntities matching an AppliedToGroup
+// for standalone selectors or corresponding to a ClusterGroup.
+func (n *NetworkPolicyController) getAppliedToWorkloads(g *antreatypes.AppliedToGroup) ([]*v1.Pod, []*v1alpha2.ExternalEntity) {
+	// Check if an internal Group object exists corresponding to this AppliedToGroup.
+	intGroup, found, _ := n.internalGroupStore.Get(g.Name)
+	if found {
+		var pods []*v1.Pod
+		ig := intGroup.(*antreatypes.Group)
+		// Generate the list of Pods based on the GroupMembers set in the internal Group.
+		for _, gm := range ig.GroupMembers {
+			// TODO: These reads can be avoided if we structure internal Group to include
+			// PodRef including NodeName, IPs and Ports and then add filter funcs to remove
+			// IPs for AppliedToGroups and NodeNames for AddressGroups.
+			pod, err := n.podLister.Pods(gm.Pod.Namespace).Get(gm.Pod.Name)
```

Comment:
gm.Pod can be `nil` here if the CG has only externalEntitySelector. Could we add in the nil checks and generalize this to support ExternalEntity here?

Reply:
have not added EE to CG yet, so it is not an issue.. i was planning to add it to CG and update in the separate PR

Reply:
Got it